probes/hasApprovedLicense/def.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: hasApprovedLicense
+short: Check that the project has a license file of approved format. An approved format is a file named either LICENSE, LICENCE, COPYING or COPYRIGHT with the file extension .html, .txt, or .md.
+motivation: >
+  A license can give users information about how the source code may or may not be used. The lack of a license will impede any kind of security review or audit and creates a legal risk for potential users.
+implementation: >
+ The implementation checks whether a license file is present and is of an approved format
+outcome:
+  - If a license file is found and is of an approved format, probe returns a single OutcomePositive (1).
+  - If a license file is missing or is not of an approved format, probe returns a single OutcomeNegative (0).
+remediation:
+  effort: Low
+  text:
+    - Update the license file format in the Github repository to be of an approved format.
```

Comment:
can we provide links on the license types that are approved?

Reply:
Added links in https://github.com/ossf/scorecard/pull/3465/commits/4d4a924c0133fbb2477e48e81159a82617a9c5fb.